docs/high/dataset.rst

```diff
@@ -504,14 +504,27 @@ Reference
 
     .. method:: astype(dtype)
 
-        Return a wrapper allowing you to read data as a particular
+        Return a read-only view allowing you to read data as a particular
         type.  Conversion is handled by HDF5 directly, on the fly::
 
             >>> dset = f.create_dataset("bigint", (1000,), dtype='int64')
-            >>> out = dset.astype('int16')[:]
+            >>> out = dset.astype('int16')[:500]
             >>> out.dtype
             dtype('int16')
 
+        This can be faster than reading the data and then
+        converting it with NumPy:
+
+            >>> out = dset[:500].astype('int16')  # Typically slower
+
+        In case of variable-width strings, calling ``.astype('T')``
+        (NumPy's native variable-width strings) is zero cost and
```

Comment:
I'm not sure what 'zero cost' means here - clearly there is still a cost to reading and repacking data. Maybe we can just say it's more efficient than making Python string objects?